src/main/java/com/salesforce/dataloader/mapping/Mapper.java

```diff
@@ -173,19 +173,28 @@ protected boolean hasMappings() {
         return this.map.values();
     }
 
+    public Collection<String> getDestColumns() {
+        return this.map.values();
+    }
+
     public String getMapping(String srcName) {
-        if(map.containsKey(srcName)) {
+        return getMapping(srcName, false);
+    }
+
+    public String getMapping(String srcName, boolean strictMatching) {
+        if (map.containsKey(srcName)) {
             return map.get(srcName);
         }
         // handle aggregate queries
-        for(Entry<String, String> entry: map.entrySet()) {
-            if(entry.getKey().endsWith("." + srcName)) {
-                return entry.getValue();
+        if (!strictMatching) {
```

Comment:
using strictMatching to limit the change only to mapping page scenario.   The root cause of the bug is because one field name, say "xyz"  and another field "abc.xyz", during refresh, it will think xyz is already mapped, when it see *.xyz.